app/src/main/java/com/vci/vectorcamapp/main/presentation/MainViewModel.kt

```diff
@@ -64,18 +67,21 @@ class MainViewModel @Inject constructor(
         }
     }
 
-    private fun determineStartDestination() {
+    private fun observeDevice() {
```

Comment:
observeDevice here would shadow the cache function deviceCache.observeDevice. Rename to something else.